### PR TITLE
GH-3: Added test framework & initial tests

### PR DESCRIPTION
**Pull request by austin-wants-to-code**
Added Jest (test framework), react-test-library (test library) and
react-test-renderer (test library) to the project. Updated & added
configs to support this. Updated to support "test" and "test-watch"
commands. Added basic tests for App.js (including a snapshot test).
Commit includes other misc changes, like comments to existing files & new
constants file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
+# 0.0.4 [2020-11-14]
+
+## Added
+  - The Jest test framework (test runner, assertion library & mocking library)
+    - This was chosen over other frameworks as it is a comprehensive test-framework rather than being limited to handling only one or two functions (i.e. being a test runner, e.g. mocha or handling mocking, e.g. sinon) 
+  - The react-test-library for overall better testing of React components 
+    - This was chosen over other React test-libraries (such as Enzyme) due to its recommendation from the ReactJS Team and its core principal of encouraging tests to resemble how the software will actually be used rather than have them rely on internal details such as state. Also, I have used Enzyme before, and have been frustrated many times from certain parts of the API
+  - The react-test-renderer library for more specific testing scenarios, like snapshot tests
+  - Jest config for running tests, reporting code coverage and ignoring the index file
+  - jsconfig file for IDEs' intellisense with Jest
```
Comment (Austin-Miller-1) [inline CHANGELOG.md:10]: wording

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+  - Basic tests for App.js
+  - Application constants file
+  - Comments to index.js & App.js
+
+## Changed
+  - "test" and "test-watch" command in package.json updated to now work with Jest
+    - Updated README with details
```
Comment (austin-wants-to-code) [inline CHANGELOG.md]: wording

File: CHANGELOG.md (hunk 1 above)
Comment (austin-wants-to-code) [inline CHANGELOG.md]: wording

File: README.md [hunk 3]
```diff
@@ -10,6 +10,10 @@ Entry point is the App.js component.
 
 Commands:
 
+- npm run test
+    - Runs the application tests. Coverage reports can be found in the "coverage" folder at the root of the project.
+- npm run test-watch
+    - Runs the applications tests and watches for changes. When a file is updated, the tests relying on this file are rerun. Coverage reports can be found in the "coverage" folder at the root of the project.
```
Comment (austin-wants-to-code) [inline README.md]: wording

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  // testResultsProcessor: undefined,
+
+  // This option allows use of a custom test runner
+  // testRunner: "jasmine2",
+
+  // This option sets the URL for the jsdom environment. It is reflected in properties such as location.href
+  // testURL: "http://localhost",
+
+  // Setting this value to "fake" allows the use of fake timers for functions such as "setTimeout"
+   timers: "modern",
```
Comment (austin-wants-to-code) [inline jest.config.js]: spacing

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+
+  // The directory where Jest should output its coverage files
+  coverageDirectory: "coverage",
+
+  // An array of regexp pattern strings used to skip coverage collection
+  coveragePathIgnorePatterns: [
+    "\\\\node_modules\\\\",
+    "index.js"
```
Comment (austin-wants-to-code) [inline jest.config.js:21]: need comment on why

File: package.json [hunk 6]
```diff
@@ -27,11 +27,14 @@
     "@babel/preset-env": "7.12.1",
     "@babel/preset-react": "7.12.1",
     "@babel/register": "7.12.1",
+    "@testing-library/react": "^11.1.2",
```
Comment (austin-wants-to-code) [inline package.json]: auto increment not allowed

File: src/components/App.js [hunk 7]
```diff
@@ -1,5 +1,8 @@
 import React, {Component} from 'react';
 
+/**
+ * Root level React component. Hosts the application.
```
Comment (austin-wants-to-code) [inline src/components/App.js]: wording

File: src/constants/applicationConstants.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+/**
+ * File contains globally available constants used by the application.
+ * Not all constants will be located here, as it may make more sense to 
+ * have certain constants stored separate from this file.
```
Comment (austin-wants-to-code) [inline src/constants/applicationConstants.js]: wording

File: src/constants/applicationConstants.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+/**
+ * File contains globally available constants used by the application.
+ * Not all constants will be located here, as it may make more sense to 
+ * have certain constants stored separate from this file. 
+ */
+
+/** Class used by application's root element */
```
Comment (austin-wants-to-code) [inline src/constants/applicationConstants.js]: wording

File: src/index.js [hunk 10]
```diff
@@ -1,5 +1,13 @@
+/**
+ * index.js is the entryway of the application. It is what connects the the React
```
Comment (austin-wants-to-code) [inline src/index.js]: wording (two "the"s)

File: src/index.js [hunk 11]
```diff
@@ -1,5 +1,13 @@
+/**
+ * index.js is the entryway of the application. It is what connects the the React 
+ * application to the actual HTML DOM. It should serve no purpose other than that.
+ * 
+ *** Because the file has just this one purpose, it should not be extended needlessly. *** 
+ *** Any potential changes to index.js should instead be made to App.js or other files ***
```
Comment (austin-wants-to-code) [inline src/index.js]: add reasoning to comment

File: jest.config.js (hunk 5 above)
Comment (austin-wants-to-code) [reply jest.config.js:21]: fixed

File: CHANGELOG.md (hunk 2 above)
Comment (austin-wants-to-code) [reply CHANGELOG.md]: fixed

File: README.md (hunk 3 above)
Comment (austin-wants-to-code) [reply README.md]: fixed

File: jest.config.js (hunk 4 above)
Comment (austin-wants-to-code) [reply jest.config.js]: fixed

File: CHANGELOG.md (hunk 1 above)
Comment (austin-wants-to-code) [reply CHANGELOG.md]: fixed

File: src/components/App.js (hunk 7 above)
Comment (austin-wants-to-code) [reply src/components/App.js]: fixed

File: src/constants/applicationConstants.js (hunk 8 above)
Comment (austin-wants-to-code) [reply src/constants/applicationConstants.js]: fixed

File: src/constants/applicationConstants.js (hunk 9 above)
Comment (austin-wants-to-code) [reply src/constants/applicationConstants.js]: fixed

File: src/index.js (hunk 10 above)
Comment (austin-wants-to-code) [reply src/index.js]: fixed

File: src/index.js (hunk 11 above)
Comment (austin-wants-to-code) [reply src/index.js]: fixed

File: package.json (hunk 6 above)
Comment (austin-wants-to-code) [reply package.json]: fixed

**Pull request merged**